src/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -311,6 +314,25 @@ public int Timeout
             }
         }
 
+        public ServicePoint ServicePoint
+        {
+            get
+            {
+                CheckHostAndPort();
+                if (_servicePoint == null)
+                {
+                    // This differs from desktop, where it uses an internal overload of FindServicePoint that just
+                    // takes a string host and an int port, bypassing the need for a Uri. We workaround that here by
+                    // creating an http Uri, simply for the purposes of getting an appropriate ServicePoint instance.
+                    // This has some subtle impact on behavior, e.g. the returned ServicePoint's Address property will
+                    // be usable, whereas in desktop it throws an exception that "This property is not supported for
+                    // protocols that do not use URI."
+                    _servicePoint = ServicePointManager.FindServicePoint(new Uri("http://" + _host + ":" + _port));
```

Comment:
It would look less odd when querying the ServicePoint properties, such as [ConnectionName](https://msdn.microsoft.com/en-us/library/system.net.servicepoint.connectionname(v=vs.110).aspx) that should return "http" now.